modules/electrophysiology_browser/php/models/ElectrophysioFile.class.inc

```diff
@@ -0,0 +1,85 @@
+<?php declare(strict_types=1);
+/**
+ * Contains the ElectrophysioFile class used for electrophysiological browser
+ *
+ * PHP Version 7
+ *
+ * @category Electrophysiology
+ * @package  Electrophysiology
+ * @author   Loris Team <loris.mni@bic.mni.mcgill.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+
+namespace LORIS\electrophysiology_browser\Models;
+
+/**
+ * This object represents an electrophysiology file in the physiological_file table.
+ *
+ * @category Electrophysiology
+ * @package  Electrophysiology
+ * @author   Loris Team <loris.mni@bic.mni.mcgill.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class ElectrophysioFile implements \LORIS\Data\DataInstance
```

Comment:
I don't get why this PR makes a distinction between a "ElectrophysiologyBrowserRowProvisioner" DataInstance and a "ElectrophysioFile" DataInstance (which serializes to JSON as the empty string?). There seems to be one row per file in the table. Why not implement toJSON on the ElectrophysioFile and use that as the type of DataInstance returned by the provisioner?

It seems to be taking the hack that was used to add models to existing data tables and using that as the "right" way of doing things rather than writing it from scratch to re-use the models for both the dataframework and the code.

Reply:
GOUP BY